modules/core/map/js/farmOS.map.behaviors.input.js

```diff
@@ -1,11 +1,56 @@
 (function () {
   farmOS.map.behaviors.input = {
     attach: function (instance) {
+
+      // Get the data input form element.
+      var input = instance.map.getTargetElement().parentElement.querySelector('[data-map-geometry-field]')
+
+      // When features change in the edit layer, write WKT to the input form element.
       instance.editAttached && instance.editAttached.then(() => {
         instance.edit.wktOn('featurechange', function(wkt) {
-          instance.map.getTargetElement().parentElement.querySelector('[data-map-geometry-field]').value = wkt;
+          input.value = wkt;
         });
       });
+
+      // Run a handleInput() callback when input changes and attempt to
+      // import the new WKT.
+      input.oninput = handleInput;
+      function handleInput(e) {
+        if (input.value) {
+
```

Comment:
Reverse conditional and return early so reader doesn't need to track as much indentation for the rest of the function?

```suggestion
        // If the value is empty, only clear features from the edit layer.
        if (!input.value) {
          instance.edit.layer.getSource().clear();
          return;
        }
```

Reply:
Good idea! Fixup: https://github.com/farmOS/farmOS/pull/670/commits/34a609e0547cf53e5b5fffcf6cad546238327760